x-pack/plugins/security_solution/server/lib/telemetry/tasks/endpoint.ts

```diff
@@ -212,7 +212,15 @@ export function createTelemetryEndpointTaskConfig(maxTelemetryBatch: number) {
             }
           }
 
-          const { cpu, memory, uptime } = endpoint.endpoint_metrics.Endpoint.metrics;
+          const {
+            cpu,
+            memory,
+            uptime,
+            documents_volume: documentsVolume,
+            malicious_behavior_rules: maliciousBehaviorRules,
+            system_impact: systemImpact,
```

Comment:
This all looks good to me. What I would say is that I noticed that from your sample document, if I removed:

  - documents_volume
  - malicious_behavior_rules
  - system_impact
  - threads

The document is `2.2KB` in size when minified, and with the surrounding ES document structure / telemetry metadata removed and minified, the size has increased to `18.8KB` - an ~x8.5 increase. Some super quick back of the envelope calcs:

10MB (10,000KB) constraint on the google cloud gateway
assume `2.2KB` is current avg
assume `20KB` is new avg after this PR is merged
`10,000 / 2.2 = 4545.45`
`10,000 / 20 = 500`

It seems that we batch @ 1,000 https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/server/lib/telemetry/constants.ts#L12

This would mean if we batch @ 1K we would always get a `413` response and lose the batch. I reckon we should reduce this down to around ~300 now to give a margin of safety. What do you think? Maybe you want to revisit this in your upcoming PR? Regardless, I think we should load test this at some point - it's overdue to be honest. 

I'm curious if you or @ferullo know if the payload sections can get much bigger than this?

Reply:
Yeah I think we'd definitely need to cut down the size of the batches given these new changes.